ui/src/components/navbar.tsx

```diff
@@ -87,6 +91,25 @@ export class Navbar extends Component<any, NavbarState> {
     }
 
     WebSocketService.Instance.getSite();
+
+    this.handleSearchParam = this.handleSearchParam.bind(this);
+  }
+
+  handleSearchParam(i: Navbar, event: any) {
+    i.state.searchParam = event.target.value;
```

Comment:
the bind would only be necessary if you wanted to use this.state as opposed to `i.state` here.

Reply:
done